### ISSUE TITLE
bug(@react-aria/selection): keyboard navigation doesn't work as expected when selectOnFocus is on and selectMode is multiple

### ISSUE THREAD

**Issue by alirezamirian** [adobe/react-spectrum#1842]
# 🐛 Bug Report 

In a selectable collection, if `selectOnFocus` is true, selection should change when navigating with the keyboard. It currently works fine when selectionMode is "single", but for doesn't for multi select collections. 

## 🤔 Expected Behavior
if `selectOnFocus` is true, navigation with keyboard should always set the selection instead of only setting the focus, and it should not depend on selection mode. A few examples of this expected behavior:
- Native html select behavior in multiple mode
- Mac's finder
- List views and tree views in [Java Swing](https://www.javatpoint.com/java-swing) (Intellij IDEs for example, or you can check out `ListDemo` in [Swing examples](https://docs.oracle.com/javase/tutorial/uiswing/examples/components/index.html))

## 😯 Current Behavior

Currently, there is an unnecessary check for selection mode here in https://github.com/adobe/react-spectrum/blob/3a1f15494c22fb9f06de7285f8b920f46f959fac/packages/%40react-aria/selection/src/useSelectableCollection.ts#L124
Which makes it broken in the multi select mode.

## 💁 Possible Solution

Remove check for single selection mode when handling `selectOnFocus`.
Regarding usages in the existing packages, the only place `selectOnFocus` can be `true` is in `useTabList`, which only supports single select mode. So there won't be any breaking change with regard to existing packages in react spectrum libraries.

## 🔦 Context

I'm trying to implement a list which behaves similar to examples mentioned above, which has `selectOnFocus` beahvior and support multi and single select modes.

## 💻 Code Sample

https://codesandbox.io/s/react-forked-r8lee?file=/src/App.tsx

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| @react-aria/selection   |  3.4.0
| Browser          | Not relevant
| Operating System | Not relevant

**Comment by alirezamirian** [comment]
Working on a PR
